File: server/app/schemas/message.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class MessageBase(BaseModel):
    session_id: int
    role: str
    content: str


class MessageCreate(BaseModel):
    content: str


class MessageResponse(MessageBase):
    id: int
    status: int
    created_at: datetime
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True

File: server/internal/schema/message.go
package schema

import "time"

type MessageCreate struct {
	Content string `json:"content" binding:"required"`
}

type MessageResponse struct {
	ID              int64     `json:"id"`
	SessionID       int64     `json:"session_id"`
	Role            string    `json:"role"`
	Content         string    `json:"content"`
	Status          int       `json:"status"`
	HasInteractions int       `json:"has_interactions"`
	CreatedAt       time.Time `json:"created_at"`
	UpdatedAt       time.Time `json:"updated_at"`
}
